### PR TITLE
Init the resource exporter code

### PR DESCRIPTION
**Pull request by huone1**
Signed-off-by: huone1 <huwanxing@huawei.com>

### CODE REVIEW

File: Makefile.def [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+# If tag not explicitly set in users default to the git sha.
+TAG ?= latest
```
Comment (wpeng102) [inline Makefile.def]: suggest follow volcano's rule, use `TAG ?= $(shell git rev-parse --verify HEAD)`

File: pkg/numatopo/update.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			},
+			Spec: v1alpha1.NumatopoSpec{
+				Policies:    GetPolicy(),
+				ResReserved: GetResReserved(),
+				NumaResMap:  GetAllResAllocatableInfo(),
+				CPUDetail:   GetCpusDetail(),
+			},
+		}
+
+		_, err = client.NodeinfoV1alpha1().Numatopologies("default").Create(context.TODO(), numaInfo, metav1.CreateOptions{})
```
Comment (wpeng102) [inline pkg/numatopo/update.go]: suggest change Numatopologies to  cluster level

**Comment by william-wang** [comment]
@huone1 
As the repo name is changed to "resource-exportor". We might support more kind of resource in the future.
So I'd suggest to restruct the directory, put cpu topology under directory/package.

File: Makefile.def (hunk 1 above)
Comment (huone1) [reply Makefile.def]: fixed

File: pkg/numatopo/update.go (hunk 2 above)
Comment (huone1) [reply pkg/numatopo/update.go]: Modified

**Comment by huone1** [comment]
> @huone1
> As the repo name is changed to "resource-exportor". We might support more kind of resource in the future.
> So I'd suggest to restruct the directory, put cpu topology under directory/package.

@william-wang  currently， keep the directory struct。just restruct the directory when adding more kind of resource

File: README.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+## Quick Start Guide
+
+### Compile
+```
+   make image [TAG=XXX]
+```
+
+### Install
+
+#### 1. Edit the file ./installer/numa-topo.yaml
```
Comment (william-wang) [inline README.md:23]: please describe that the configuration is optional or mandatory.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+````
+   There are some options which you can use to configure the deployment
```
Comment (william-wang) [inline README.md]: add description to tell user why you need the path of these file.

File: installer/numa-topo.yaml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        name: resource-topology
+    spec:
+      serviceAccountName: resource-exporter-account
+      containers:
+        - name: resource-topology
+          image: volcanosh/numatopo:latest
+          imagePullPolicy: IfNotPresent
+          args:
+            - --logtostderr
+            - --kubelet-conf=/host/kubeletconf/config.yaml
```
Comment (william-wang) [inline installer/numa-topo.yaml:63]: "/host/kubeletconf/config.yaml " is not a valid path, suggest change it to "/var/lib/kubelet/config.yaml"

File: pkg/numatopo/interface.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+	"volcano.sh/resource-exporter/pkg/args"
+)
+
+// NumaInfo is the interface of resource topolgy data
+type NumaInfo interface {
+	Name() string
+	Update(opt *args.Argument) NumaInfo
+	GetResourceInfoMap() v1alpha1.ResourceInfo
+	GetCPUDetail() map[string]v1alpha1.CPUInfo
```
Comment (shinytang6) [inline pkg/numatopo/interface.go]: l think this func should be more general

File: main.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+		return nil, err
+	}
+
+	return versioned.NewForConfigOrDie(config), err
+}
+
+func numatopoIsExist(client *versioned.Clientset) (bool, error) {
+	hostname := os.Getenv("MY_NODE_NAME")
+	if hostname == "" {
+		return false, fmt.Errorf("get Hostname failed")
```
Comment (william-wang) [inline main.go:53]: Is the log support rotating? if not, the error log will occupy too much disk.

Comment (huone1) [reply main.go:53]: currently， the log isn't redirected to the specific path，just use "kubectl logs XXX" to check it, So if users want it, they  need to  adapt the rotation.

File: pkg/numatopo/interface.go (hunk 6 above)
Comment (huone1) [reply pkg/numatopo/interface.go]: use GetResTopoDetail() interface{} to replace it

File: README.md (hunk 3 above)
Comment (huone1) [reply README.md:23]: reedit it， described in different parts

File: README.md (hunk 4 above)
Comment (huone1) [reply README.md]: added  the description for per option,

**Review by shinytang6 — APPROVED** [review]

**Review by wpeng102 — APPROVED** [review]

**Pull request merged**